resources/ad_applications.go

```diff
@@ -0,0 +1,610 @@
+			},
+			{
+				Name:        "homepage",
+				Description: "The home page of the application",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "identifier_uris",
+				Description: "A collection of URIs for the application",
+				Type:        schema.TypeStringArray,
+			},
+			{
+				Name:        "informational_urls_terms_of_service",
+				Description: "The terms of service URI",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("InformationalUrls.TermsOfService"),
+			},
+			{
+				Name:        "informational_urls_marketing",
+				Description: "The marketing URI",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("InformationalUrls.Marketing"),
+			},
+			{
+				Name:        "informational_urls_privacy",
+				Description: "The privacy policy URI",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("InformationalUrls.Privacy"),
+			},
+			{
+				Name:        "informational_urls_support",
+				Description: "The support URI",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("InformationalUrls.Support"),
+			},
+			{
+				Name:        "is_device_only_auth_supported",
+				Description: "Specifies whether this application supports device authentication without a user The default is false",
+				Type:        schema.TypeBool,
+			},
+			{
+				Name:        "known_client_applications",
+				Description: "Client applications that are tied to this resource application Consent to any of the known client applications will result in implicit consent to the resource application through a combined consent dialog (showing the OAuth permission scopes required by the client and the resource)",
+				Type:        schema.TypeStringArray,
+			},
+			{
+				Name:        "logout_url",
+				Description: "the url of the logout page",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("LogoutURL"),
+			},
+			{
+				Name:        "oauth2_allow_implicit_flow",
+				Description: "Whether to allow implicit grant flow for OAuth2",
+				Type:        schema.TypeBool,
+			},
+			{
+				Name:        "oauth2_allow_url_path_matching",
+				Description: "Specifies whether during a token Request Azure AD will allow path matching of the redirect URI against the applications collection of replyURLs The default is false",
+				Type:        schema.TypeBool,
+				Resolver:    schema.PathResolver("Oauth2AllowURLPathMatching"),
+			},
+			{
+				Name:        "oauth2_require_post_response",
+				Description: "Specifies whether, as part of OAuth 20 token requests, Azure AD will allow POST requests, as opposed to GET requests The default is false, which specifies that only GET requests will be allowed",
+				Type:        schema.TypeBool,
+			},
+			{
+				Name:        "org_restrictions",
+				Description: "A list of tenants allowed to access application",
+				Type:        schema.TypeStringArray,
+			},
+			{
+				Name:     "optional_claims",
```

Comment:
I decided to represent this field as a JSON, because otherwise it creates three relations.